### ISSUE TITLE
Can't Return Cards from Hand to Top of Library

### ISSUE THREAD

**Issue by dmmyccnt** [Cockatrice/Cockatrice#4501]
**System Information:**
Client Version: 2.8.1-beta.4 (2021-11-04)
PictureLoader: [card:  "Mountain"  set:  "NEO" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Mountain"  set:  "NEO" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Leonin Relic-Warder"  set:  "MB1" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Leonin Relic-Warder"  set:  "MB1" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Gold Myr"  set:  "TD2" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Gold Myr"  set:  "TD2" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Memnite"  set:  "PLIST" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Memnite"  set:  "PLIST" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Master's Call"  set:  "PW11" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Master's Call"  set:  "PW11" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Viridian Claw"  set:  "TD2" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Viridian Claw"  set:  "TD2" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Ardent Recruit"  set:  "TD2" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Ardent Recruit"  set:  "TD2" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Arrest"  set:  "MB1" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Arrest"  set:  "MB1" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Whispersilk Cloak"  set:  "MB1" ]: Picture found on disk.
PictureLoader: [card:  "Iron Myr"  set:  "FMB1" ]: Trying to load picture
PictureLoader: [card:  "Iron Myr"  set:  "FMB1" ]: Picture found on disk.
PictureLoader: [card:  "Signal Pest"  set:  "PW11" ]: Trying to load picture
PictureLoader: [card:  "Whispersilk Cloak"  set:  "MB1" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Whispersilk Cloak"  set:  "MB1" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Signal Pest"  set:  "PW11" ]: Picture found on disk.
PictureLoader: [card:  "Siege Mastodon"  set:  "M20" ]: Trying to load picture
PictureLoader: [card:  "Siege Mastodon"  set:  "M20" ]: Picture found on disk.
PictureLoader: [card:  "Concussive Bolt"  set:  "MBS" ]: Trying to load picture
PictureLoader: [card:  "Concussive Bolt"  set:  "MBS" ]: Picture found on disk.
PictureLoader: [card:  "Origin Spellbomb"  set:  "SOM" ]: Trying to load picture
PictureLoader: [card:  "Origin Spellbomb"  set:  "SOM" ]: Picture found on disk.
PictureLoader: [card:  "Peace Strider"  set:  "2XM" ]: Trying to load picture
PictureLoader: [card:  "Peace Strider"  set:  "2XM" ]: Picture found on disk.
PictureLoader: [card:  "Accorder Paladin"  set:  "MBS" ]: Trying to load picture
PictureLoader: [card:  "Accorder Paladin"  set:  "MBS" ]: Picture found on disk.
PictureLoader: [card:  "Mountain"  set:  "NEO" ]: Trying to load picture
PictureLoader: [card:  "Mountain"  set:  "NEO" ]: Picture found on disk.
PictureLoader: [card:  "White Sun's Zenith"  set:  "CMR" ]: Trying to load picture
PictureLoader: [card:  "White Sun's Zenith"  set:  "CMR" ]: Picture found on disk.
PictureLoader: [card:  "Leonin Relic-Warder"  set:  "MB1" ]: Trying to load picture
PictureLoader: [card:  "Leonin Relic-Warder"  set:  "MB1" ]: Picture found on disk.
PictureLoader: [card:  "Gold Myr"  set:  "TD2" ]: Trying to load picture
PictureLoader: [card:  "Gold Myr"  set:  "TD2" ]: Picture found on disk.
PictureLoader: [card:  "Myrsmith"  set:  "2XM" ]: Trying to load picture
PictureLoader: [card:  "Myrsmith"  set:  "2XM" ]: Picture found on disk.
PictureLoader: [card:  "Kuldotha Ringleader"  set:  "MBS" ]: Trying to load picture
PictureLoader: [card:  "Kuldotha Ringleader"  set:  "MBS" ]: Picture found on disk.
PictureLoader: [card:  "Razorfield Rhino"  set:  "MBS" ]: Trying to load picture
PictureLoader: [card:  "Razorfield Rhino"  set:  "MBS" ]: Picture found on disk.
PictureLoader: [card:  "Memnite"  set:  "PLIST" ]: Trying to load picture
PictureLoader: [card:  "Memnite"  set:  "PLIST" ]: Picture found on disk.
PictureLoader: [card:  "Galvanic Blast"  set:  "2XM" ]: Trying to load picture
PictureLoader: [card:  "Galvanic Blast"  set:  "2XM" ]: Picture found on disk.
PictureLoader: [card:  "Master's Call"  set:  "PW11" ]: Trying to load picture
PictureLoader: [card:  "Master's Call"  set:  "PW11" ]: Picture found on disk.
PictureLoader: [card:  "Viridian Claw"  set:  "TD2" ]: Trying to load picture
PictureLoader: [card:  "Viridian Claw"  set:  "TD2" ]: Picture found on disk.
PictureLoader: [card:  "Silvercoat Lion"  set:  "M13" ]: Trying to load picture
PictureLoader: [card:  "Silvercoat Lion"  set:  "M13" ]: Picture found on disk.
PictureLoader: [card:  "Victory's Herald"  set:  "CM2" ]: Trying to load picture
PictureLoader: [card:  "Victory's Herald"  set:  "CM2" ]: Picture found on disk.
PictureLoader: [card:  "Leonin Skyhunter"  set:  "DDG" ]: Trying to load picture
PictureLoader: [card:  "Leonin Skyhunter"  set:  "DDG" ]: Picture found on disk.
PictureLoader: [card:  "Ardent Recruit"  set:  "TD2" ]: Trying to load picture
PictureLoader: [card:  "Ardent Recruit"  set:  "TD2" ]: Picture found on disk.
PictureLoader: [card:  "Arrest"  set:  "MB1" ]: Trying to load picture
PictureLoader: [card:  "Arrest"  set:  "MB1" ]: Picture found on disk.
PictureLoader: [card:  "Whispersilk Cloak"  set:  "MB1" ]: Trying to load picture
PictureLoader: [card:  "Whispersilk Cloak"  set:  "MB1" ]: Picture found on disk.
addCounter: "Player 1" 0 "life"
addCounter: "Player 1" 1 "w"
addCounter: "Player 1" 2 "u"
addCounter: "Player 1" 3 "b"
addCounter: "Player 1" 4 "r"
addCounter: "Player 1" 5 "g"
addCounter: "Player 1" 6 "x"
addCounter: "Player 1" 7 "storm"
addCounter: "Player 2" 0 "life"
addCounter: "Player 2" 1 "w"
addCounter: "Player 2" 2 "u"
addCounter: "Player 2" 3 "b"
addCounter: "Player 2" 4 "r"
addCounter: "Player 2" 5 "g"
addCounter: "Player 2" 6 "x"
addCounter: "Player 2" 7 "storm"
addCounter: "Player 1" 0 "life"
addCounter: "Player 1" 1 "w"
addCounter: "Player 1" 2 "u"
addCounter: "Player 1" 3 "b"
addCounter: "Player 1" 4 "r"
addCounter: "Player 1" 5 "g"
addCounter: "Player 1" 6 "x"
addCounter: "Player 1" 7 "storm"
addCounter: "Player 2" 0 "life"
addCounter: "Player 2" 1 "w"
addCounter: "Player 2" 2 "u"
addCounter: "Player 2" 3 "b"
addCounter: "Player 2" 4 "r"
addCounter: "Player 2" 5 "g"
addCounter: "Player 2" 6 "x"
addCounter: "Player 2" 7 "storm"
PictureLoader: cache fail for "_trice_card_back_6897"
QPixmap::scaled: Pixmap is a null pixmap
PictureLoader: cache fail for "_trice_card_back_437596"
QPixmap::scaled: Pixmap is a null pixmap
PictureLoader: [card:  "Gold Myr"  set:  "TD2" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Gold Myr"  set:  "TD2" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Gold Myr"  set:  "TD2" ]: Trying to load picture
PictureLoader: [card:  "Gold Myr"  set:  "TD2" ]: Picture found on disk.
PictureLoader: [card:  "Galvanic Blast"  set:  "2XM" ]: Trying to load picture
PictureLoader: cache fail for "_trice_card_back_6897"
PictureLoader: [card:  "Galvanic Blast"  set:  "2XM" ]: Picture found on disk.
PictureLoader: [card:  "Plains"  set:  "NEO" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Plains"  set:  "NEO" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Plains"  set:  "NEO" ]: Trying to load picture
PictureLoader: [card:  "Plains"  set:  "NEO" ]: Picture found on disk.
PictureLoader: [card:  "Mountain"  set:  "NEO" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Mountain"  set:  "NEO" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Mountain"  set:  "NEO" ]: Trying to load picture
PictureLoader: [card:  "Iron Myr"  set:  "FMB1" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Iron Myr"  set:  "FMB1" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Mountain"  set:  "NEO" ]: Picture found on disk.
PictureLoader: [card:  "Iron Myr"  set:  "FMB1" ]: Trying to load picture
PictureLoader: [card:  "Iron Myr"  set:  "FMB1" ]: Picture found on disk.
PictureLoader: [card:  "Ardent Recruit"  set:  "TD2" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Ardent Recruit"  set:  "TD2" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Ardent Recruit"  set:  "TD2" ]: Trying to load picture
PictureLoader: [card:  "Ardent Recruit"  set:  "TD2" ]: Picture found on disk.
PictureLoader: cache fail for "_trice_card_back_437613"
PictureLoader: [card:  "Mountain"  set:  "NEO" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Mountain"  set:  "NEO" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Mountain"  set:  "NEO" ]: Trying to load picture
PictureLoader: [card:  "Mountain"  set:  "NEO" ]: Picture found on disk.
PictureLoader: [card:  "Plains"  set:  "NEO" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Plains"  set:  "NEO" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Plains"  set:  "NEO" ]: Trying to load picture
PictureLoader: [card:  "Plains"  set:  "NEO" ]: Picture found on disk.
PictureLoader: [card:  "Iron Myr"  set:  "FMB1" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Iron Myr"  set:  "FMB1" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Iron Myr"  set:  "FMB1" ]: Trying to load picture
PictureLoader: [card:  "Iron Myr"  set:  "FMB1" ]: Picture found on disk.
PictureLoader: [card:  "Ardent Recruit"  set:  "TD2" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Ardent Recruit"  set:  "TD2" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Ardent Recruit"  set:  "TD2" ]: Trying to load picture
PictureLoader: [card:  "Ardent Recruit"  set:  "TD2" ]: Picture found on disk.
PictureLoader: [card:  "Plains"  set:  "NEO" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Plains"  set:  "NEO" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Plains"  set:  "NEO" ]: Trying to load picture
PictureLoader: [card:  "Mountain"  set:  "NEO" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Mountain"  set:  "NEO" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Plains"  set:  "NEO" ]: Picture found on disk.
PictureLoader: [card:  "Mountain"  set:  "NEO" ]: Trying to load picture
PictureLoader: [card:  "Mountain"  set:  "NEO" ]: Picture found on disk.
PictureLoader: [card:  "Gold Myr"  set:  "TD2" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Gold Myr"  set:  "TD2" ]: Requested set property ( "muid" ) for Url template ( "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Gold Myr"  set:  "TD2" ]: Trying to load picture
PictureLoader: [card:  "Gold Myr"  set:  "TD2" ]: Picture found on disk.
PictureLoader: [card:  "Galvanic Blast"  set:  "2XM" ]: Trying to load picture
PictureLoader: [card:  "Galvanic Blast"  set:  "2XM" ]: Picture found on disk.

_______________________________________________________________________________________

When playing a local game, whenever either player tries to return a card from their hand to the top of the library, no matter which card is selected to return to the top of the library, it's always the card that is left-most in the hand that is returned to the top. If any other card in hand is selected to return to the top of the library, that card stays in hand and it can't be moved to any other zone for the rest of the game. This happens whether dragging the card back to the top of the library with the mouse, or right clicking on the card and selecting "Move to -> Top of Library."

In short, whenever I try to return almost any card from my hand to the top of my library, that card gets frozen in my hand and can't be played or moved to any other zone.

This happens every single time.
_______________________________________________________________________________________

**Steps to reproduce:**
 - Start a local game
 - Draw any number of cards
 - Try to return any card in your hand other than the left-most card to the top of the library

**Comment by ebbit1q** [comment]
this issue has already been fixed since then in https://github.com/Cockatrice/Cockatrice/pull/4491

I'll publish a new beta that includes this fix
